File: #1310.xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {

        int size = arr.size();
        vector<int> dp = arr;
        for (int i = 1; i < size; i++) {
            dp[i] ^= dp[i-1];
        }
        
        int left;
        int right;
        vector<int> results;
        for (vector<int>& query : queries) {
            left = query[0];
            right = query[1];
            int result = dp[right];
            if (left > 0) {
                result ^= dp[left - 1];
            }
            results.push_back(result);
        }
        return results;
    }
};

File: #1310.xor-queries-of-a-subarray.js
/**
 * @param {number[]} arr
 * @param {number[][]} queries
 * @return {number[]}
 */
var xorQueries = function(arr, queries) {
    const xors = arr.reduce((result, num, index) => {
        result.push(result[index] ^ num);
        return result;
    }, [0]);

    return queries.map(([left, right]) => {
        return xors[right + 1] ^ xors[left];
    });
};
